### ISSUE TITLE
Build system does not find SDL headers on vanilla Arch Linux

### ISSUE THREAD

**Issue by amno1** [intel/fastuidraw#39]
I have sdl installed but build system does not see includes out of the box. Headers in Arch Linux are in /usr/include/SDL . I modified some "base" makefile to include it in include paths and I got one demo to go through, but then next example complains again :-). Which makefile is proper place to add sdl so the path is included in all demos, lib etc?

**Comment by krogueintel** [comment]
That is... peculiar.  First off, do you have SDL or SDL*2* installed?  The build system uses sdl2-config to determine where the SDL2 headers are. It is possible, but incredibly unlikely, that Arch Linux does not install sdl2-config.

For reference, the demos build rules are defined in make/Makefile.demos.rules.mk.

I will submit a commit shortly to have the build system have a build target that checks if the requirements are met.

**Comment by krogueintel** [comment]
I've added some more build magic:
- System exclusively uses pkg-config to get headers and libs (before sdl2-config was used for SDL2)
- System checks for SDL2_image (via pkg-config)
- New build target "check" that checks (according to pkg-config) if build dependencies are met and checks if GL/GLES headers exists

Run make check and paste the output here so I can help you resolve the building issue.

**Comment by krogueintel** [comment]
Did the latest commit resolve your issues?

**Comment by amno1** [comment]
I am really sorry for late replay, I wasn't much at home last two days, and I didn't really have time to check on github. 

Than you very much for your effort. Yes it does work after latest commit. I have just cloned new and was able to build. I must admit that it seems that I have missed to install one package. I had both sdl and sdl2 installed, however I didn't had sd2_image. I didn't even know that package existed. After I looked into the Makefile you mention above I saw what I was missing, I installed it and it works out of the box.

I was little in a rush the other day, so I skimmed through your make files but completeley missed what exactly you are doing there.

I was able to run some demos, it seems to work fine. Thank you for help.
